tests/integration/test_mqtt.py

```diff
@@ -92,12 +94,64 @@ def callback(msg):
         result["event"].set()
 
     tx = MqttTransport(HOST)
-    topic = Topic("/messages_compas_eve_test/", TestMessage)
+    topic = Topic("/messages_compas_eve_test/test_message_type_parsing/", TestMessage)
 
     Subscriber(topic, callback, transport=tx).subscribe()
     Publisher(topic, transport=tx).publish(TestMessage(name="Jazz"))
 
     received = result["event"].wait(timeout=3)
     assert received, "Message not received"
     assert result["value"].name == "Jazz"
+    assert result["value"]["name"] == "Jazz", "Messages should be accessible as dict"
     assert result["value"].hello_name == "Hello Jazz"
+
+
+def test_compas_data_as_message():
+
+    class DataTestMessage(Data):
+        def __init__(self, name=None, eclipse=None):
+            self.name = name
+            self.eclipse = eclipse
+
+        def __to_data__(self):
```

Comment:
ah is this not a re-implementation of `__data__`?

Reply:
sorry, i was wrong again, it should be `__data__` indeed